### ISSUE TITLE
Does Kubernetes actually work now with Clear Linux?

### ISSUE THREAD

**Issue by KyleSanderson** [clearlinux/cloud-native-setup#319]
I followed the old instructions and it was just... busted. Completely random versions of k8s and cri-o were in the pkg tree that were totally incompatible with each-other led me to deploy to production with docker-ce from previous metal.

I see the instructions now have completely changed which gives me great hope, but I may be too far down the rabbit hole now to test. https://docs.01.org/clearlinux/latest/tutorials/kubernetes.html

In short: Does Kubernetes actually work now with Clear Linux? Has anyone done an installation?

**Comment by eadamsintel** [comment]
It does work, and we do basic testing to ensure the stack comes up. Is there a particular version of Clear Linux where you said it is not working? The scripts in this repo are to help make it easier to setup Clear Linux but are really tested and intended for a fresh install of Clear Linux.  For example, these scripts assume that only one CRI (usually CRIO) is running and `kubeadm` will complain about multiple cri sockets if you have Docker running at the same time.  You can either stop Docker or edit the `kubeadm.yaml` 

We recently updated the manual Kubernetes instructions that you linked above, and those instructions follow the `create_stack.sh` script.  

It shouldn't be the case that incompatible k8s and crio versions are together in the same release of Clear Linux. If that happens then we want to fix that in the next build.

**Comment by inmanturbo** [comment]
> Does Kubernetes actually work now with Clear Linux? Has anyone done an installation?

Minicube works well. So does Kubernetes. The instructions don't really lead to a meaningful end though. The docs are light and the usage is difficult to compare against other distros. In short: Yes it will does work, but it takes some sweat, and some serious chops ;)

I tried following the instructions, and ended up just going with minicube on a local instance, then using another distro entirely in production. It's not that it doesn't work well, but the lack of documentation makes it time consuming and difficult to use. No worse than aws I suppose, but nothing like gcloud with GKE.

**Comment by MarkusAGSchmidt** [comment]
I just tried the kubernetes distribution k3s on Clear Linux and it was a disappointing!
The basic communication (node-to-node, local pods) was working but the non-local pod communication failed. There was no way to communicate with a pod running on the Clear Linux node. Pods that were running on Ubuntu or Raspberry nodes were working fine. I tried many days to figure out the problem with Clear Linux and one can find a lot of information (e.g. rp_filter) but nothing worked.
I wonder how Clear Linux achieved the Kubernetes certification?

My working environment is:
Kubernetes: v1.18.6+k3s1
CNI: Flannel with VXLAN backend
Nodes: Ubuntu 18.04.4 LTS , Raspbian GNU/Linux 10 (buster)

And the problem node runs on Clear Linux 33590.

**Comment by KyleSanderson** [comment]
The whole thing is a joke 🤣 I raised this issue with the CNCF https://github.com/cncf/k8s-conformance/pull/1118 and dankhon asserted this is as designed.

**Comment by CraigSterrett** [comment]
@MarkusAGSchmidt @KyleSanderson The Kubernetes conformance program provides assurance that the Kubernetes installation included with Clear Linux and as described in the README uploaded with the conformance PR https://github.com/cncf/k8s-conformance/tree/master/v1.17/clearlinux passes all of the conformance requirements. This does not guarantee that any Kubernetes installation will run on Clear Linux without issue. We have not validated that the k3s release of Kubernetes works on Clear Linux.
